src/libraries/System.Text.Json/tests/JsonTestHelper.cs

```diff
@@ -814,5 +814,57 @@ public static string NormalizeLineEndings(this string value)
             => s_replaceNewlines ?
             value.Replace(CompiledNewline, Environment.NewLine) :
             value;
+
+        public static void AssertJsonEqual(string expected, string actual)
+        {
+            using JsonDocument expectedDom = JsonDocument.Parse(expected);
+            using JsonDocument actualDom = JsonDocument.Parse(actual);
+            AssertJsonEqual(expectedDom.RootElement, actualDom.RootElement);
+        }
+
+        private static void AssertJsonEqual(JsonElement expected, JsonElement actual)
+        {
+            JsonValueKind valueKind = expected.ValueKind;
+            Assert.Equal(valueKind, actual.ValueKind);
+
+            switch (valueKind)
```

Comment:
As more tests use these helper methods, we'll need to add more cases here; e.g. `case JsonValueKind.Number`.